src/main/scala/com/atomist/rug/kind/docker/DockerFileMutableView.scala

```diff
@@ -32,109 +44,155 @@ class DockerFileMutableView(originalBackingObject: FileArtifact, pmv: ProjectMut
       case s: String => _content.addMaintainer(s"$maintainerName <$s>")
       case _ => _content.addMaintainer(maintainerName)
     }
   }
 
-  @ExportFunction(readOnly = false, description = "Add or update LABEL directive")
+  @ExportFunction(readOnly = true,
+    description = "Return map of labels.  If there are no labels, an empty map is returned.")
+  def getLabels: Map[String, String] = _content.getLabels
+
+  @ExportFunction(readOnly = false, description = "Add or update LABEL instruction")
   def addOrUpdateLabel(@ExportFunctionParameterDescription(name = "labelContents",
-    description = "The contents of the LABEL directive") labelContents: String) {
+    description = "The contents of the LABEL instruction") labelContents: String) {
     _content.addOrUpdateLabel(labelContents)
   }
 
-  @ExportFunction(readOnly = false, description = "Add LABEL directive")
+  @ExportFunction(readOnly = false, description = "Add LABEL instruction")
   def addLabel(@ExportFunctionParameterDescription(name = "labelContents",
-    description = "The contents of the LABEL directive") labelContents: String) {
+    description = "The contents of the LABEL instruction") labelContents: String) {
     _content.addLabel(labelContents)
   }
 
-  @ExportFunction(readOnly = false, description = "Add RUN directive")
+  @ExportFunction(readOnly = true,
+    description = "Return sequence of RUN instructions as strings. If the RUN instruction is an array, a string representation of the array is included. If there are no RUN instructions, an empty sequence is returned.")
+  def getRuns: Seq[String] = Dockerfile.seqStringOrArrayToString(_content.getRuns)
+
+  @ExportFunction(readOnly = false, description = "Add RUN instruction")
   def addRun(@ExportFunctionParameterDescription(name = "runContents",
-    description = "The contents of the RUN directive") runContents: String) {
+    description = "The contents of the RUN instruction") runContents: String) {
     _content.addRun(runContents)
   }
 
-  @ExportFunction(readOnly = false, description = "Add COPY directive")
+  @ExportFunction(readOnly = true,
+    description = "Return sequence of COPY instructions as a sequence of strings. If there are no COPY instructions, an empty sequence is returned.")
+  def getCopies: Seq[Seq[String]] = _content.getCopies
+
+  @ExportFunction(readOnly = false, description = "Add COPY instruction")
   def addCopy(@ExportFunctionParameterDescription(name = "copyContents",
-    description = "The contents of the COPY directive") copyContents: String) {
+    description = "The contents of the COPY instruction") copyContents: String) {
     _content.addCopy(copyContents)
   }
 
-  @ExportFunction(readOnly = false, description = "Add ADD directive")
+  @ExportFunction(readOnly = true,
+    description = "Return sequence of ADD instructions as a sequence of strings. If there are no ADD instructions, an empty sequence is returned.")
+  def getAdds: Seq[Seq[String]] = _content.getAdds
```

Comment:
After adding the various types to the TypeScript mappings I wondered if we should limit return values to simple types like boolean and string. The pre-existing getter in DockerFileMutableView returned `Seq[Int]`, so I thought there was precedent. @russmiles mentioned the DSL might have trouble with complex return types too. Perhaps I'll just force them all into strings. The only tricky thing will be collapsing collections when they are actually needed, e.g., multiple ADD, COPY, or RUN instructions.

@johnsonr what do you think about, in general, restricting the allowed return types?